src/widgets/lv_img.c

```diff
@@ -590,6 +618,15 @@ static void lv_img_event(const lv_obj_class_t * class_p, lv_event_t * e)
         }
     }
     else if(code == LV_EVENT_GET_SELF_SIZE) {
+        if(img->cf == LV_IMG_CF_UNKNOWN) {
+            lv_point_t size = { lv_obj_get_style_width(obj, LV_PART_MAIN),
+                                lv_obj_get_style_height(obj, LV_PART_MAIN)
+                              };
+            if(size.x == LV_SIZE_CONTENT || size.y == LV_SIZE_CONTENT)
+                /*Image is not decoded yet, so let's decode it to figure out its size, if required*/
+                decode_img(img, &size);
```

Comment:
If `LV_EVENT_GET_SELF_SIZE` is called multiple times on the same source, ideally the metadata is read from the cache, right?

However if the cache is small (usually on embedded), you need to re-cache/re-open the image here multiple times. 

As here we need only the width and height, I think it'd be better to set `img->h  = -1` and `img->w = -1 ` on `src` change, and call `decode_img` only if `img->w/h` is `-1`.

I'm not sure but it seems the same "if img->w/h is -1 get the size" is required in `draw_img` too. It because when the object's size is set explicitly (not `LV_SIZE_CONTENT`),  `LV_EVENT_GET_SELF_SIZE` won't be called .

Reply:
If the image is decoded, then the `cf` field will be populated (it's 0 = unknown upon construction), so it'll skip decoding twice. 
I don't really grasp the event chain in LVGL. So you're very likely right here. I thought that `GET_SELF_SIZE` would be called unconditionally. So, I'll try to add a similar check in the `draw_img` call.

Right now, I've seen they are multiple event leading to `draw_img` being called and one is `LV_EVENT_COVER_CHECK`. Should this one require image decoding too ?